File: http/Node/lib/result.js
// Copyright Core DF
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

export function missingEnv(vars) {
  return {
    status_code: 601,
    error: `Environment variables ${vars} should be defined`,
  };
}

export function transportError(message = 'inaccessible') {
  return { status_code: 0, error: message };
}

File: http/Dart/lib/result.dart
// Copyright Core DF — Apache License 2.0
class CoreautoResult {
  static Map<String, dynamic> missingEnv(String vars) => {
    'status_code': 601,
    'error': 'Environment variables $vars should be defined',
  };
  static Map<String, dynamic> transportError([String message = 'inaccessible']) => {
    'status_code': 0,
    'error': message,
  };
}
